clients/client/src/main/java/org/projectnessie/client/StreamingUtil.java

```diff
@@ -63,25 +76,28 @@ private StreamingUtil() {
    * <p>The {@link Stream} returned by {@code getEntriesStream(ref, OptionalInt.empty())}, if not
    * limited, returns all commit-log entries.
    *
-   * @param ref a named reference (branch or tag name) or a commit-hash
+   * @param api The {@link NessieApiV1} to use
+   * @param ref a named reference (branch or tag name).
+   * @param builderCustomizer a Consumer of the {@link GetEntriesBuilder} that is passed to the
+   *     caller of this method. It allows the caller to customize the GetEntriesBuilder with
+   *     additional parameters (e.g.: hashOnRef, filter).
+   * @param maxRecords - a maximum number of records in the stream. If it is {@code
+   *     Optional.empty()} the stream will be unbounded.
    * @return stream of {@link Entry} objects
    */
   public static Stream<Entry> getEntriesStream(
       @NotNull NessieApiV1 api,
       @NotNull String ref,
-      @Nullable String hashOnRef,
-      @Nullable String filter,
-      OptionalInt maxRecords)
+      @NotNull Consumer<GetEntriesBuilder> builderCustomizer,
+      @NotNull OptionalInt maxRecords)
       throws NessieNotFoundException {
 
     return new ResultStreamPaginator<>(
             EntriesResponse::getEntries,
-            (ref1, pageSize, token) ->
-                builderWithPaging(api.getEntries(), pageSize, token)
-                    .refName(ref1)
-                    .hashOnRef(hashOnRef)
-                    .filter(filter)
-                    .get())
+            (reference, pageSize, token) -> {
+              builderCustomizer.accept(api.getEntries());
```

Comment:
Looks like this line has no effect?

Reply:
Builders are mutable, so it will modify the entries builder.